src/SixLabors.Fonts/TextLayout.cs

```diff
@@ -207,54 +203,80 @@ public IReadOnlyList<GlyphLayout> GenerateLayout(ReadOnlySpan<char> text, Render
+                        startOfLine = true;
 
-                            // move the characters to the next line
-                            for (int j = lastWrappableLocation; j < layout.Count; j++)
+                        // move the characters to the next line
+                        for (int j = lastWrappableLocation; j < layout.Count; j++)
+                        {
+                            if (layout[j].IsWhiteSpace())
                             {
-                                if (layout[j].IsWhiteSpace)
-                                {
-                                    wrappingOffset += layout[j].Width;
-                                    layout.RemoveAt(j);
-                                    j--;
-                                    continue;
-                                }
-
-                                Vector2 current = layout[j].Location;
-                                layout[j] = new GlyphLayout(layout[j].GraphemeIndex, layout[j].CodePoint, layout[j].Glyph, new Vector2(current.X - wrappingOffset, current.Y + lineHeight), layout[j].Width, layout[j].Height, layout[j].LineHeight, startOfLine, layout[j].IsWhiteSpace, layout[j].IsControlCharacter);
-                                startOfLine = false;
-
-                                location.X = layout[j].Location.X + layout[j].Width;
+                                wrappingOffset += layout[j].Width;
+                                layout.RemoveAt(j);
+                                j--;
+                                continue;
                             }
 
-                            location.Y += lineHeight;
-                            totalHeight += lineHeight;
-                            firstLine = false;
-                            lastWrappableLocation = -1;
+                            GlyphLayout current = layout[j];
+                            var wrapped = new GlyphLayout(
+                                current.GraphemeIndex,
+                                current.CodePoint,
+                                current.Glyph,
+                                new Vector2(current.Location.X - wrappingOffset, current.Location.Y + lineHeight),
+                                current.Width,
+                                current.Height,
+                                current.LineHeight,
+                                startOfLine);
+
+                            startOfLine = false;
+
+                            location.X = wrapped.Location.X + wrapped.Width;
+                            layout[j] = wrapped;
                         }
+
+                        location.Y += lineHeight;
+                        totalHeight += lineHeight;
+                        firstLine = false;
+                        lastWrappableLocation = -1;
                     }
 
                     previousGlyph = glyph;
                 }
-                else if (codepoints[i] == '\r')
+                else if (codePoint.Value == '\r')
```

Comment:
I need to review the logic here. \r on it's own could potentially be incorrectly detected on it's own since we enumerating the code points.